core/src/main/java/org/jruby/truffle/runtime/core/RubyArray.java

```diff
@@ -244,8 +216,11 @@ public int getSize() {
     }
 
     public void setSize(int size) {
+        assert verifyStore(this.store, size);
```

Comment:
Do you have a pointer to something that would better explain how this change improves startup/warmup?